modules/standard/DateTime.chpl

```diff
@@ -0,0 +1,1571 @@
+    }
+    var offset = utcoffset();
+    if tzinfo != nil {
+      var sign: string;
+      if offset.days < 0 {
+        offset = -offset;
+        sign = "-";
+      } else {
+        sign = "+";
+      }
+      ret = ret + sign + makeNDigits(2, offset.seconds/(60*60)) + ":" +
+                         makeNDigits(2, offset.seconds % (60*60) / 60);
+    }
+    return ret;
+  }
+
+  /* Return the offset from UTC */
+  proc time.utcoffset() {
+    if tzinfo == nil {
+      return new timedelta();
+    } else {
+      return tzinfo.utcoffset(datetime.today());
+    }
+  }
+
+  /* Return the daylight saving time offset */
+  proc time.dst() {
+    if tzinfo == nil {
+      return new timedelta();
+    } else {
+      return tzinfo.dst(datetime.today());
+      //return new timedelta();
+    }
+  }
+
+  /* Return the name of the timezone for this time value */
+  proc time.tzname() {
+    if tzinfo == nil then
+      return "";
+    else
+      return tzinfo.tzname(new datetime(1,1,1));
+  }
+
+  /* Return a string matching the format argument for this time */
+  proc time.strftime(fmt: string) {
+    extern proc strftime(s: c_void_ptr, size: size_t, format: c_string, ref timeStruct: tm);
+    const bufLen: size_t = 100;
+    var buf: [1..bufLen] c_char;
+    var timeStruct: tm;
+
+    timeStruct.tm_sec = second: int(32);
+    timeStruct.tm_min = minute: int(32);
+    timeStruct.tm_hour = hour: int(32);
+    timeStruct.tm_year = 1900;
+    timeStruct.tm_mday = 1;
+    timeStruct.tm_mon = 1;
+
+    if tzinfo != nil {
+      timeStruct.tm_gmtoff = abs(utcoffset()).seconds;
+      timeStruct.tm_zone = __primitive("cast", c_ptr(c_char), tzname().c_str());
+      timeStruct.tm_isdst = dst().seconds:int(32);
+    } else {
+      timeStruct.tm_gmtoff = 0;
+      timeStruct.tm_zone = __primitive("cast", c_ptr(c_char), "".c_str());
+      timeStruct.tm_isdst = -1;
+    }
+
+    strftime(c_ptrTo(buf), bufLen, fmt.c_str(), timeStruct);
+    var str = __primitive("cast", c_string, c_ptrTo(buf)): string;
+
+    return str;
+  }
+
+  /* Operators on time values */
+
+  pragma "no doc"
+  proc ==(t1: time, t2: time): bool {
+    var dt1 = datetime.combine(d=new date(2000, 1, 1), t=t1);
+    var dt2 = datetime.combine(d=new date(2000, 1, 1), t=t2);
+    return dt1 == dt2;
+  }
+
+  pragma "no doc"
+  proc !=(t1: time, t2: time) {
+    return !(t1 == t2);
+  }
+
+  pragma "no doc"
+  proc <(t1: time, t2: time): bool {
+    if (t1.tzinfo != nil && t2.tzinfo == nil) ||
+        (t1.tzinfo == nil && t2.tzinfo != nil) {
+      halt("both datetimes must both be either naive or aware");
+    } else if t1.tzinfo == t2.tzinfo {
+      const sec1 = t1.hour*3600 + t1.minute*60 +
+                   t1.second + t1.microsecond/1000000.0;
+      const sec2 = t2.hour*3600 + t2.minute*60 +
+                   t2.second + t2.microsecond/1000000.0;
+      return sec1 < sec2;
+    } else {
+      // As far as I can tell, python's datetime.time() comparisons don't
```

Comment:
In this comment, I was looking at the constructor named `time`, not the getter named `time`.  It's a little confusing because so many things have the same name in the python module.  The module is named `datetime`, there's a `datetime` type, and a `datetime` constructor, then there's the `time` type, `time` constructor and `time` getter function.